### PR TITLE
Create pytest test harness for bash integration tests

### LINKED ISSUE #1925: Add pytest test harness to execute each bash script and also geopm/service/integration/test/test_controller.py
Relates to #1861

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Relates to #1861 issue, task from github issues
- Fixes #1925 change request from github issues

Adds a test harness that uses subTest feature to programmatically add tests that execute bash scripts that match test_*.sh pattern.

### CODE REVIEW

File: service/integration/test/TestBashExamples.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            with self.subTest(bash_example=os.path.basename(script)):
+                pid = subprocess.Popen([script],
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+                out, err = pid.communicate()
+                output = f'\n\nSTDOUT:\n{out.decode("utf-8")}\n\nSTDERR:\n{err.decode("utf-8")}'
+                if pid.returncode != 0:
+                    self.assertEqual(0, pid.returncode, output)
+                else:
+                    sys.stdout.write(output)
```
Comment (cmcantalupo) [inline service/integration/test/TestBashExamples.py]:
```suggestion
                self.assertEqual(0, pid.returncode, output)
                sys.stdout.write(output)
```

Comment (cmcantalupo) [reply service/integration/test/TestBashExamples.py]: This seems better, but I will have to test.

File: service/integration/test/TestBashExamples.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def test_all(self):
+        for script in self.bash_tests:
+            script_name = os.path.basename(script)
+            with self.subTest(bash_example=script_name):
+                pid = subprocess.Popen([script],
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+                out, err = pid.communicate()
+                out = out.decode("utf-8")
+                err = err.decode("utf-8")
```
Comment (dannosliwcd) [inline service/integration/test/TestBashExamples.py]: We can have popen do the encoding/decoding for us if we tell it to handle io in text mode. If our minimum version is 3.7, we can set text=True. If not, we can use universal_newlines=True (same purpose, just a non-obvious name).

**Review by dannosliwcd — DISMISSED** [review]
LGTM. Minor suggestion doesn't affect correctness.

File: service/integration/test/TestBashExamples.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_all(self):
+        for script in self.bash_tests:
+            script_name = os.path.basename(script)
+            with self.subTest(bash_example=script_name):
+                pid = subprocess.Popen([script],
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+                out, err = pid.communicate()
+                out = out.decode("utf-8")
```
Comment (bgeltz) [inline service/integration/test/TestBashExamples.py]: We do this in the launcher:
```
stdout_encoding = locale.getpreferredencoding()
```
Then:
```
stdout.write(stdout_bytes.decode(encoding=stdout_encoding))
```

**Review by bgeltz — CHANGES_REQUESTED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**